src/http_interface.py

```diff
@@ -63,29 +65,72 @@ def __init__(self, charm: CharmBase, relation_name: str):
         observe(charm.on[relation_name].relation_departed, self._on_relation_changed)
         observe(charm.on[relation_name].relation_broken, self._on_relation_broken)
 
+    @abc.abstractmethod
     def _on_relation_joined(self, _: RelationJoinedEvent) -> None:
-        """Abstract method to handle relation-joined event."""
+        """Abstract method to handle relation-joined event.
 
+        Raises:
+            NotImplementedError: if the abstract method is not implemented.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
     def _on_relation_changed(self, _: RelationChangedEvent) -> None:
-        """Abstract method to handle relation-changed event."""
+        """Abstract method to handle relation-changed event.
 
+        Raises:
+            NotImplementedError: if the abstract method is not implemented.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
     def _on_relation_broken(self, _: RelationBrokenEvent) -> None:
-        """Abstract method to handle relation-changed event."""
+        """Abstract method to handle relation-changed event.
+
+        Raises:
+            NotImplementedError: if the abstract method is not implemented.
+        """
+        raise NotImplementedError
 
     @property
     def relations(self) -> list[Relation]:
         """The list of Relation instances associated with the charm."""
-        return list(self.charm.model.relations[self.relation_name])
+        relations = self.charm.model.relations.get(self.relation_name)
+        if relations is None:
+            return []
```

Comment:
`.get(self.relation_name, [])`

Reply:
Thanks, I updated the code to reflect this